From 2c29116e2fc1f564354203dcf19f8bf901b420b6 Mon Sep 17 00:00:00 2001
From: Priyanka Jain <priyanka.jain@nxp.com>
Date: Mon, 31 Aug 2020 13:22:03 +0530
Subject: [PATCH 3/9] board/freescale: Add board support for db1046

db1046 board is similar to ls1046awrdb with few differences
 I2C devices and addresses
 no RGMII
 no SGMII

Signed-off-by: Pankit Garg <pankit.garg@nxp.com>
Signed-off-by: Priyanka Jain <priyanka.jain@nxp.com>
---
 arch/arm/Kconfig                              | 18 ++++++
 arch/arm/cpu/armv8/Kconfig                    |  1 +
 .../armv8/fsl-layerscape/fsl_lsch2_speed.c    | 10 ++-
 board/freescale/ls1046awrdb/Kconfig           |  2 +-
 board/freescale/ls1046awrdb/MAINTAINERS       |  1 +
 board/freescale/ls1046awrdb/eth.c             |  9 ++-
 configs/db1046_tfa_defconfig                  | 61 +++++++++++++++++++
 include/configs/ls1046awrdb.h                 |  6 ++
 8 files changed, 103 insertions(+), 5 deletions(-)
 create mode 100644 configs/db1046_tfa_defconfig

diff --git a/arch/arm/Kconfig b/arch/arm/Kconfig
index 25912249ba..a009a26248 100644
--- a/arch/arm/Kconfig
+++ b/arch/arm/Kconfig
@@ -1511,6 +1511,24 @@ config TARGET_LS1046AWRDB
 	  development platform that supports the QorIQ LS1046A
 	  Layerscape Architecture processor.
 
+config TARGET_DB1046
+	bool "Support db1046"
+	select ARCH_LS1046A
+	select ARM64
+	select ARMV8_MULTIENTRY
+	select ARCH_SUPPORT_TFABOOT
+	select BOARD_EARLY_INIT_F
+	select BOARD_LATE_INIT
+	select DM_SPI_FLASH if DM_SPI
+	select SUPPORT_SPL
+	select FSL_DDR_BIST
+	select FSL_DDR_INTERACTIVE if !SPL
+	imply SCSI
+	help
+	  Support for NXP DB1046 Platform.
+	  The DB1046 is a development platform that supports
+	  the QorIQ LS1046A Layerscape Architecture processor.
+
 config TARGET_H2200
 	bool "Support h2200"
 	select CPU_PXA
diff --git a/arch/arm/cpu/armv8/Kconfig b/arch/arm/cpu/armv8/Kconfig
index a38a9aa35e..e25ca077c3 100644
--- a/arch/arm/cpu/armv8/Kconfig
+++ b/arch/arm/cpu/armv8/Kconfig
@@ -108,6 +108,7 @@ config PSCI_RESET
 		   !TARGET_LS1043ARDB && !TARGET_LS1043AQDS && \
 		   !TARGET_LS1046ARDB && !TARGET_LS1046AQDS && \
 		   !TARGET_LS1046AFRWY && !TARGET_LS1046AWRDB && \
+		   !TARGET_DB1046 && \
 		   !TARGET_LS2081ARDB && !TARGET_LX2160ARDB && \
 		   !TARGET_LX2160AQDS && \
 		   !ARCH_UNIPHIER && !TARGET_S32V234EVB
diff --git a/arch/arm/cpu/armv8/fsl-layerscape/fsl_lsch2_speed.c b/arch/arm/cpu/armv8/fsl-layerscape/fsl_lsch2_speed.c
index d199054a7d..a12f542c0b 100644
--- a/arch/arm/cpu/armv8/fsl-layerscape/fsl_lsch2_speed.c
+++ b/arch/arm/cpu/armv8/fsl-layerscape/fsl_lsch2_speed.c
@@ -130,6 +130,7 @@ void get_sys_info(struct sys_info *sys_info)
 #define HWA_CGA_M2_CLK_SHIFT	0
 #if defined(CONFIG_TARGET_LS1046ARDB) || \
 	defined(CONFIG_TARGET_LS1046AWRDB) || \
+	defined(CONFIG_TARGET_DB1046) || \
 	defined(CONFIG_TARGET_LS1043ARDB)
 	rcw_tmp = in_be32(&gur->rcwsr[15]);
 	switch ((rcw_tmp & HWA_CGA_M2_CLK_SEL) >> HWA_CGA_M2_CLK_SHIFT) {
@@ -137,7 +138,8 @@ void get_sys_info(struct sys_info *sys_info)
 		sys_info->freq_cga_m2 = freq_c_pll[1];
 		break;
 #if defined(CONFIG_TARGET_LS1046ARDB) || \
-	defined(CONFIG_TARGET_LS1046AWRDB)
+	defined(CONFIG_TARGET_LS1046AWRDB) ||\
+	defined(CONFIG_TARGET_DB1046)
 	case 2:
 		sys_info->freq_cga_m2 = freq_c_pll[1] / 2;
 		break;
@@ -146,7 +148,8 @@ void get_sys_info(struct sys_info *sys_info)
 		sys_info->freq_cga_m2 = freq_c_pll[1] / 3;
 		break;
 #if defined(CONFIG_TARGET_LS1046ARDB) || \
-	defined(CONFIG_TARGET_LS1046AWRDB)
+	defined(CONFIG_TARGET_LS1046AWRDB) ||\
+	defined(CONFIG_TARGET_DB1046)
 	case 6:
 		sys_info->freq_cga_m2 = freq_c_pll[0] / 2;
 		break;
@@ -192,7 +195,8 @@ int get_clocks(void)
 #ifdef CONFIG_FSL_ESDHC
 #if defined(CONFIG_FSL_ESDHC_USE_PERIPHERAL_CLK)
 #if defined(CONFIG_TARGET_LS1046ARDB) || \
-	defined(CONFIG_TARGET_LS1046AWRDB)
+	defined(CONFIG_TARGET_LS1046AWRDB) ||\
+	defined(CONFIG_TARGET_DB1046)
 	gd->arch.sdhc_clk = sys_info.freq_cga_m2 / 2;
 #endif
 #if defined(CONFIG_TARGET_LS1043ARDB)
diff --git a/board/freescale/ls1046awrdb/Kconfig b/board/freescale/ls1046awrdb/Kconfig
index ecd716ab4b..3e19d5ab14 100644
--- a/board/freescale/ls1046awrdb/Kconfig
+++ b/board/freescale/ls1046awrdb/Kconfig
@@ -1,5 +1,5 @@
 
-if TARGET_LS1046AWRDB
+if TARGET_LS1046AWRDB || TARGET_DB1046
 
 config SYS_BOARD
 	default "ls1046awrdb"
diff --git a/board/freescale/ls1046awrdb/MAINTAINERS b/board/freescale/ls1046awrdb/MAINTAINERS
index a1fa7fad2c..89a768254f 100644
--- a/board/freescale/ls1046awrdb/MAINTAINERS
+++ b/board/freescale/ls1046awrdb/MAINTAINERS
@@ -6,3 +6,4 @@ F:	board/freescale/ls1046awrdb/ls1046awrdb.c
 F:	include/configs/ls1046awrdb.h
 F:	configs/ls1046awrdb_tfa_defconfig
 F:	configs/ls1046awrdb_tfa_SECURE_BOOT_defconfig
+F:	configs/db1046_tfa_defconfig
diff --git a/board/freescale/ls1046awrdb/eth.c b/board/freescale/ls1046awrdb/eth.c
index f2b376fcd7..2a69892f9b 100644
--- a/board/freescale/ls1046awrdb/eth.c
+++ b/board/freescale/ls1046awrdb/eth.c
@@ -15,8 +15,10 @@
 int board_eth_init(bd_t *bis)
 {
 #ifdef CONFIG_FMAN_ENET
+#ifndef CONFIG_TARGET_DB1046
 	int i;
 	struct memac_mdio_info dtsec_mdio_info;
+#endif
 	struct memac_mdio_info tgec_mdio_info;
 	struct mii_dev *dev;
 	u32 srds_s1;
@@ -26,6 +28,7 @@ int board_eth_init(bd_t *bis)
 			FSL_CHASSIS2_RCWSR4_SRDS1_PRTCL_MASK;
 	srds_s1 >>= FSL_CHASSIS2_RCWSR4_SRDS1_PRTCL_SHIFT;
 
+#ifndef CONFIG_TARGET_DB1046
 	dtsec_mdio_info.regs =
 		(struct memac_mdio_controller *)CONFIG_SYS_FM1_DTSEC_MDIO_ADDR;
 
@@ -33,6 +36,7 @@ int board_eth_init(bd_t *bis)
 
 	/* Register the 1G MDIO bus */
 	fm_memac_mdio_init(bis, &dtsec_mdio_info);
+#endif
 
 	tgec_mdio_info.regs =
 		(struct memac_mdio_controller *)CONFIG_SYS_FM1_TGEC_MDIO_ADDR;
@@ -41,13 +45,14 @@ int board_eth_init(bd_t *bis)
 	/* Register the 10G MDIO bus */
 	fm_memac_mdio_init(bis, &tgec_mdio_info);
 
+#ifndef CONFIG_TARGET_DB1046
 	/* Set the two on-board RGMII PHY address */
 	fm_info_set_phy_address(FM1_DTSEC3, RGMII_PHY1_ADDR);
 
 	/* Set the two on-board SGMII PHY address */
 	fm_info_set_phy_address(FM1_DTSEC5, SGMII_PHY1_ADDR);
 	fm_info_set_phy_address(FM1_DTSEC6, SGMII_PHY2_ADDR);
-
+#endif
 	/* Set the on-board AQ PHY address */
 	fm_info_set_phy_address(FM1_10GEC2, FM1_10GEC2_PHY_ADDR);
 
@@ -60,9 +65,11 @@ int board_eth_init(bd_t *bis)
 		break;
 	}
 
+#ifndef CONFIG_TARGET_DB1046
 	dev = miiphy_get_dev_by_name(DEFAULT_FM_MDIO_NAME);
 	for (i = FM1_DTSEC1; i < FM1_DTSEC1 + CONFIG_SYS_NUM_FM1_DTSEC; i++)
 		fm_info_set_mdio(i, dev);
+#endif
 
 	/* XFI on MAC 10 */
 	dev = miiphy_get_dev_by_name(DEFAULT_FM_TGEC_MDIO_NAME);
diff --git a/configs/db1046_tfa_defconfig b/configs/db1046_tfa_defconfig
new file mode 100644
index 0000000000..8e8a2ca770
--- /dev/null
+++ b/configs/db1046_tfa_defconfig
@@ -0,0 +1,61 @@
+CONFIG_ARM=y
+CONFIG_TARGET_DB1046=y
+CONFIG_TFABOOT=y
+CONFIG_SYS_TEXT_BASE=0x82000000
+CONFIG_QSPI_AHB_INIT=y
+CONFIG_NR_DRAM_BANKS=2
+CONFIG_ARMV8_SEC_FIRMWARE_SUPPORT=y
+CONFIG_SEC_FIRMWARE_ARMV8_PSCI=y
+CONFIG_AHCI=y
+CONFIG_DISTRO_DEFAULTS=y
+CONFIG_FIT_VERBOSE=y
+CONFIG_OF_BOARD_SETUP=y
+CONFIG_BOOTDELAY=10
+CONFIG_USE_BOOTARGS=y
+CONFIG_BOOTARGS="console=ttyS0,115200 root=/dev/ram0 earlycon=uart8250,mmio,0x21c0500 mtdparts=1550000.spi-0:1m(rcw),15m(u-boot),48m(kernel.itb);7e800000.flash:16m(nand_uboot),48m(nand_kernel),448m(nand_free)"
+CONFIG_MISC_INIT_R=y
+CONFIG_CMD_GPT=y
+CONFIG_CMD_I2C=y
+CONFIG_CMD_MMC=y
+CONFIG_CMD_NAND=y
+CONFIG_CMD_PCI=y
+CONFIG_CMD_SF=y
+CONFIG_CMD_USB=y
+CONFIG_CMD_CACHE=y
+CONFIG_MP=y
+CONFIG_MTDPARTS_DEFAULT="mtdparts=1550000.spi-0:1m(rcw),15m(u-boot),48m(kernel.itb);7e800000.flash:16m(nand_uboot),48m(nand_kernel),448m(nand_free)"
+CONFIG_OF_CONTROL=y
+CONFIG_DEFAULT_DEVICE_TREE="fsl-ls1046a-wrdb"
+CONFIG_ENV_IS_IN_MMC=y
+CONFIG_ENV_IS_IN_SPI_FLASH=y
+CONFIG_DM=y
+CONFIG_SATA_CEVA=y
+CONFIG_FSL_CAAM=y
+CONFIG_DM_MMC=y
+CONFIG_FSL_ESDHC=y
+CONFIG_SPI_FLASH=y
+# CONFIG_SPI_FLASH_BAR is not set
+CONFIG_SPI_FLASH_STMICRO=y
+# CONFIG_SPI_FLASH_USE_4K_SECTORS is not set
+CONFIG_PHYLIB=y
+CONFIG_PHY_AQUANTIA=y
+CONFIG_PHY_GIGE=y
+CONFIG_E1000=y
+CONFIG_FMAN_ENET=y
+CONFIG_PCI=y
+CONFIG_DM_PCI=y
+CONFIG_DM_PCI_COMPAT=y
+CONFIG_PCIE_LAYERSCAPE=y
+CONFIG_SYS_NS16550=y
+CONFIG_SPI=y
+CONFIG_DM_SPI=y
+CONFIG_FSL_QSPI=y
+CONFIG_USB=y
+CONFIG_DM_USB=y
+CONFIG_USB_XHCI_HCD=y
+CONFIG_USB_XHCI_DWC3=y
+CONFIG_EFI_LOADER_BOUNCE_BUFFER=y
+CONFIG_DM_I2C=y
+CONFIG_DM_RTC=y
+CONFIG_RTC_PCF2127=y
+CONFIG_DM_GPIO=y
diff --git a/include/configs/ls1046awrdb.h b/include/configs/ls1046awrdb.h
index 713d809949..0dba8aeed1 100644
--- a/include/configs/ls1046awrdb.h
+++ b/include/configs/ls1046awrdb.h
@@ -29,7 +29,11 @@
 #define CONFIG_ID_EEPROM
 #define CONFIG_SYS_I2C_EEPROM_NXID
 #define CONFIG_SYS_EEPROM_BUS_NUM		0
+#ifdef CONFIG_TARGET_DB1046
+#define CONFIG_SYS_I2C_EEPROM_ADDR		0x52
+#else
 #define CONFIG_SYS_I2C_EEPROM_ADDR		0x57
+#endif
 #define CONFIG_SYS_I2C_EEPROM_ADDR_LEN		1
 #define CONFIG_SYS_EEPROM_PAGE_WRITE_BITS	3
 #define CONFIG_SYS_EEPROM_PAGE_WRITE_DELAY_MS	5
@@ -88,10 +92,12 @@
 #define CONFIG_ENV_ADDR CONFIG_SYS_FSL_QSPI_BASE + CONFIG_ENV_OFFSET
 
 /* FMan */
+#ifndef CONFIG_TARGET_DB1046
 #define RGMII_PHY1_ADDR			0x1
 
 #define SGMII_PHY1_ADDR			0x2
 #define SGMII_PHY2_ADDR			0x3
+#endif
 
 #define FM1_10GEC2_PHY_ADDR		0x0
 
-- 
2.17.1

